homeassistant/components/mystrom/switch.py

```diff
@@ -28,14 +33,30 @@
 )
 
 
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the myStrom entities."""
+    info = hass.data[DOMAIN][entry.entry_id]["info"]
+    device_type = info["type"]
+
+    if device_type in [101, 106, 107]:
+        device = hass.data[DOMAIN][entry.entry_id]["device"]
+        try:
+            await device.get_state()
+        except MyStromConnectionError as err:
+            _LOGGER.error("No route to myStrom plug: %s", info["ip"])
+            raise ConfigEntryNotReady() from err
+        async_add_entities([MyStromSwitch(device)])
+
+
 async def async_setup_platform(
```

Comment:
`async_setup_platform` should not create entities.
It should instead trigger an import and raise a `deprecated_yaml` issue in the issue registry.

Reply:
Implemented